llama2.mojo

```diff
@@ -250,12 +250,17 @@ struct FileBuf:
         self.offset = 0
         self.size = 0
 
-    fn move_offset(inout self, size: Int):
-        self.offset += size
+    fn move_offset(inout self, size: Int) raises:
+        let new_offset = self.offset + size
+        if new_offset > self.size:
+            raise Error("Attempting to read past the end of a FileBuf")
```

Comment:
move_offset looks fine, generally, but I think the error should raise when we're actually attempting to data, as the message suggest